pkg/bucket/aws.go

```diff
@@ -170,3 +173,18 @@ func (a awsBucketClient) Delete() (bool, error) {
 
 	return true, nil
 }
+
+// StripDefaultPorts removes port 80 from HTTP URLs and 443 from HTTPS URLs.
+// Defer to the actual AWS SDK implementation to match its behavior exactly.
+func StripDefaultPorts(fromUrl string) (string, error) {
+	u, err := url.Parse(fromUrl)
+	if err != nil {
+		return "", err
+	}
+	r := http.Request{
+		URL: u,
+	}
+	request.SanitizeHostForHeader(&r)
+	r.URL.Host = r.Host
+	return r.URL.String(), nil
+}
```

Comment:
LGTM but I would move this to more generic place.. maybe util.go you never know when this might be needed for non aws.

Reply:
Okay, I tried moving this to pkg/common.